queries/idris/highlights.scm

```diff
@@ -0,0 +1,218 @@
+; ------------------------------------------------------------------------------
+; Literals and comments
+[
+  (integer)
+  (quantity)
+] @number
+
+(literal
+  (number)) @number.float
+
+(char) @character
+
+[
+  (string)
+  (pat_string)
+  (triple_quote_string)
+] @string
+
+(comment) @comment
+
+((comment) @comment.documentation
+  (#lua-match? @comment.documentation "^|||"))
+
+(unit) @constant
+
+; more general captures are moved to the top
+; before overwritten later by more specific captures
+(loname) @variable
+
+; ------------------------------------------------------------------------------
+; Punctuation
+[
+  "("
+  ")"
+  "{"
+  "@{"
+  "}"
+  "["
+  "[<"
+  "]"
+] @punctuation.bracket
+
+[
+  (comma)
+  (colon)
+  (pat_op)
+  (tuple_operator)
+] @punctuation.delimiter
+
+(pat_name
+  (loname) @variable.parameter)
+
+; ------------------------------------------------------------------------------
+; Types
+(signature
+  (loname) @type)
+
+; ------------------------------------------------------------------------------
+; Keywords, operators, imports
+[
+  "if"
+  "then"
+  "else"
+  "case"
+  "of"
+] @keyword.conditional
+
+(module) @module
```

Comment:
Maybe delete this, and instead add this below (with other module highlights):

```query
(module
  (caname) @module)
```
that way the highlights are more granular